flocker/node/agents/test/test_blockdevice.py

```diff
@@ -122,10 +158,79 @@ def test_no_devices(self):
         """
         self.assertDiscoveredState(self.deployer, [])
 
+    def test_attached_unmounted_device(self):
+        """
+        If a volume is attached but not mounted, it is included as a
+        non-manifest dataset returned by ``BlockDeviceDeployer.discover_state``
+        and not as a manifestation on the ``NodeState``.
+        """
+        unmounted = self.api.create_volume(
+            dataset_id=uuid4(),
+            size=REALISTIC_BLOCKDEVICE_SIZE,
+        )
+        self.api.attach_volume(
+            unmounted.blockdevice_id, self.expected_hostname
+        )
+        self.assertDiscoveredState(self.deployer, [], [unmounted.dataset_id])
```

Comment:
The signature of the `assertDiscoveredState` isn't obvious. This diff would have been easier to read if we used keyword arguments for `expected_manifestations` and `expected_nonmanifest_datasets` here and in later tests.